Swift Sources/UIViewController+Promise.swift

```diff
@@ -19,6 +20,15 @@ class MFMailComposeViewControllerProxy: NSObject, MFMailComposeViewControllerDel
         }
         PMKRelease(self)
     }
+
+    func messageComposeViewController(controller: MFMessageComposeViewController!, didFinishWithResult result: MessageComposeResult) {
+        if result.value == MessageComposeResultFailed.value {
+            controller.reject(NSError(domain: PMKErrorDomain, code: 4, userInfo: [NSLocalizedDescriptionKey: "The user’s attempt to save or send the message was unsuccessful."]))
```

Comment:
4 is `PMKAccessDeniedError`, so we should use the symbol and not a magic number. Still it seems the wrong code. We may need a new code. It is bizarre that this part of MessageUI doesn't provide an `NSError`! I vote for `PMKOperationFailed` which is as generic sounding as Apple should be embarrassed.